code/mob/living/critter/skeleton.dm

```diff
@@ -81,5 +106,76 @@
 		HH.icon_state = "handr"
 
 	setup_healths()
-		add_hh_flesh(50, 1)
-		add_hh_flesh_burn(50, 0.7)
+		add_hh_flesh(src.health_brute, src.health_brute_vuln)
+		add_hh_flesh_burn(src.health_burn, src.health_brute_vuln)
+
+	seek_target(var/range = 6)
+		. = list()
+		for (var/mob/living/C in hearers(range, src))
+			if (isdead(C)) continue
+			if (isintangible(C)) continue //don't attack what you can't touch
+			if (istype(C, /mob/living/critter/skeleton)) continue
+			if (iswizard(C) && src.wizardSpawn) continue
+			. += C
+
+	critter_attack(mob/target)
+		var/datum/targetable/critter/tackle = src.abilityHolder.getAbility(/datum/targetable/critter/tackle)
+		if (!tackle.disabled && tackle.cooldowncheck())
+			tackle.handleCast(target)
+		else
+			return ..()
+
+	death(var/gibbed)
+		if (prob(src.revivalChance))
+			..()
+			src.revivalChance -= src.revivalDecrement
+			SPAWN(rand(400,800))
+				src.full_heal()
+				src.set_density(1)
```

Comment:
the density setting seems unnecessary assuming it's handled in lifeprocesses or similar.
Is this to guarantee they're immediately dense, or just a legacy thing that got pasted over?